src/BloomExe/Edit/EditingModel.cs

```diff
@@ -607,6 +615,11 @@ void OnIdleAfterDocumentSupposedlyCompleted(object sender, EventArgs e)
 			// listen for events raised by javascript
 			_view.AddMessageEventListener("saveAccordionSettingsEvent", SaveAccordionSettings);
 			_view.AddMessageEventListener("setModalStateEvent", SetModalState);
+			// Currently this event is raised by the JS which is run when we are about to change pages.
+			// If it is an origami page, we we indeed need to do this origami cleanup.
+			// However, even if it is not an origami page, we need to call RethinkPageAndReloadIt, because
+			// it also has the crtical side effect of saving the page. Something should probably be renamed
+			// (or the responsibilities divided) to reflect this, but I can't see a clear way forward.
```

Comment:
I think it is because all this happens BEFORE we change page. The Refresh is probably mostly async and gets aborted by switching the browser to the new page. Still, better not to do it in this case. Maybe we should make a new event for this, with most of the code common.